common/lib/container-definitions/src/error.ts

```diff
@@ -36,14 +38,30 @@ export enum ContainerErrorType {
 /**
  * Base interface for all errors and warnings at container level
  */
-export interface IErrorBase {
+export interface IErrorBase extends Partial<Error> {
```

Comment:
Given that we keep adding same things to IFluidErrorBase & IErrorBase, I have hard time reasoning why there needs to be two almost identical types that have no direct relationship to each other. Should at least one inherit from another and clearly spell out in comments when one should expect one over another?

Reply:
Yeah see #6980.

Reply:
I am still pondering yours and Tony's feedback on fluidErrorCode etc.  For now I'm content to have IFluidErrorBase be our internal interface and IErrorBase be the public API.  I'm actively thinking about this though.